### PR TITLE
expand 'Making good pull requests' section (SOFTWARE-5459)

### PR DESCRIPTION
**Pull request by edquist**

### CODE REVIEW

File: docs/software/git-software-development.md [hunk 1]
```diff
@@ -163,3 +151,137 @@ A release of a software is created from your local clone of a software project.
...
+being introduced into the codebase.
+Having to review several ideas that were put in to and then taken out of the
+changeset is a distraction, and makes it harder to see what the end result is
+for the new changes.
+
+If you have such work-in-progress commits, first combine them (this is also
+called "squashing" or "rebasing"), and then break them up into logically
+distinct commits as necessary, representing the final changeset.
+As mentioned above, one way to do this is with a combination of `git rebase -i`
+and `git gui`, though there are other third-party tools available also.
```
Comment (brianhlin) [inline docs/software/git-software-development.md:211]:
```suggestion
and `git gui`, though there are other third-party tools (e.g., magit) available also.
```

I can't help myself

**Review by matyasselmeci — APPROVED** [review]
I like it!

I'll make sure my review message fits in an 80-column window.

File: docs/software/git-software-development.md [hunk 2]
```diff
@@ -163,3 +151,137 @@ A release of a software is created from your local clone of a software project.
...
+by a blank line plus the message body.)
+
+You can also explain in the commit message body _how_ this commit accomplishes
+the stated purpose in the subject - and you may find yourself needing to do
+this if there are some tricky details in the changes.
+But even if it is perfectly clear from the code and your commit message _what_
+you are changing and _how_ you are going about it, it is not always clear
+_why_ the change is needed or desired - so it is important to explain your
+reasons, in order to make this clear to the reviewer.
+
```
Comment (brianhlin) [inline docs/software/git-software-development.md:241]: An example would be nice. Not to toot my own horn but I'm proud of this commit body https://github.com/opensciencegrid/topology/commit/c3524138ac8d46eee2a3c33cb75fac50acab41c4

File: docs/software/git-software-development.md [hunk 3]
```diff
@@ -163,3 +151,137 @@ A release of a software is created from your local clone of a software project.
...
+#### Reference any relevant tickets
+
+Code changes often are related to a JIRA ticket, for instance SOFTWARE-1234.
```
Comment (brianhlin) [inline docs/software/git-software-development.md:244]:
```suggestion
Code changes often are related to a Jira ticket, for instance SOFTWARE-1234.
```

They rebranded

File: docs/software/git-software-development.md [hunk 4]
```diff
@@ -163,3 +151,137 @@ A release of a software is created from your local clone of a software project.
...
+the interested reader.
+
+See online guides such as
+[this one](https://dev.to/ruanbrandao/how-to-make-good-git-commits-256k)
+for more details.
+
+#### Brownie points
+
+You will get brownie points from Carl, personally, if you strive to make your
+code (and other text files) fit within an 80-column terminal window.
```
Comment (brianhlin) [inline docs/software/git-software-development.md:285]: Cute. I'll allow it.

**Review by brianhlin — APPROVED** [review]
This is awesome! Some comments, take em or leave em

**Comment by brianhlin** [comment]
Just to note it somewhere, we discussed on Zoom about adding a subsection about how to do good reviews, e.g. batching comments into a single review

File: docs/software/git-software-development.md (hunk 3 above)
Comment (edquist) [reply docs/software/git-software-development.md:244]: Oh man i missed the memo on that one.

File: docs/software/git-software-development.md (hunk 1 above)
Comment (edquist) [reply docs/software/git-software-development.md]: You got it.

File: docs/software/git-software-development.md (hunk 2 above)
Comment (edquist) [reply docs/software/git-software-development.md:242]: Sure yeah we can use that one - updated.

...

Btw speaking of tooting my own horn, i had a quick skim of the multi-line commit messages in

```
~/work/software/technology$ git log --author=edquist --no-merges | less
```

and i was not disappointed ;)

File: docs/software/git-software-development.md (hunk 3 above)
Comment (edquist) [reply docs/software/git-software-development.md]: Update.  Note that we called it `JIRA` in a couple other places in this doc, so for consistency I have updated those also.

It's still `JIRA` in a number of other docs, but I feel like it's outside the scope of this ticket to fix them all in this PR.

File: docs/software/git-software-development.md (hunk 4 above)
Comment (edquist) [reply docs/software/git-software-development.md:293]: Thanks Boss.  It wouldn't have been a legit Strong Opinions section without it ;)

**Comment by edquist** [comment]
@brianhlin - i think we're all set to merge this.

Link checker now is only complaining about broken links, not from this page:

```
Checking 285 external links...
- ./site/meetings/2020/TechArea20201130/index.html
  *  External link https://github.com/opensciencegrid/docker-software-base/runs/1437033352?check_suite_focus=true#step:7:16 failed: 404 No error
- ./site/policy/service-migrations-spring-2018/index.html
  *  External link http://collector.opensciencegrid.org/ failed: 301 SSL peer certificate or SSH remote key was not OK
- ./site/release/new-release-series/index.html
  *  internally linking to ../release/cut-sw-release, which does not exist (line 2996)
     <a href="../release/cut-sw-release">Cut a release</a>
- ./site/software/koji-workflow/index.html
  *  External link https://docs.pagure.org/koji/ failed: got a time out (response code 0)
```

I don't have admin override privs on this repo to hit the red reckless-merge button, so i'll have to leave that for you.

**Pull request merged**